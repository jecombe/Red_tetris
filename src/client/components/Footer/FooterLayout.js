import React from 'react';
import Grid from '@material-ui/core/Grid';

import FooterAuthors from './FooterAuthors';

const FooterLayout = () => (
  <Grid container justify="center" alignItems="center">
    <FooterAuthors />
  </Grid>
);


export default FooterLayout;
