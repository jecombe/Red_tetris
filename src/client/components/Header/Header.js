import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Grid from '@material-ui/core/Grid';

import HeaderLogo from './HeaderLogo';
import HeaderHomeContainer from '../../containers/Header/HeaderHomeContainer';
import HeaderStateContainer from '../../containers/Header/HeaderStateContainer';

const Header = () => (
  <AppBar position="static" color="inherit">
    <Grid container alignItems="center">
      <Grid item xs={3}>
        <HeaderHomeContainer />
      </Grid>
      <Grid item xs={6}>
        <HeaderLogo />
      </Grid>
      <Grid item xs={3}>
        <HeaderStateContainer />
      </Grid>
    </Grid>
  </AppBar>
);

export default Header;
