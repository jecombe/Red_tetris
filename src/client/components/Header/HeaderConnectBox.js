import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  connectIcon: (props) => ({
    color: props.colorState,
  }),
});

const HeaderConnectBox = (props) => {
  const { socketState, loggedState, colorState } = props;
  const classes = useStyles({ colorState });

  return (
    <Grid container justify="center" alignItems="center">
      <Grid item xs={3} container justify="center" alignItems="center">
        <FiberManualRecordIcon fontSize="small" className={classes.connectIcon} />
        <Typography color="initial">
          {socketState}
        </Typography>
      </Grid>
      <Grid item xs={5} container justify="center" alignItems="center">
        <Typography color="initial">
          {loggedState}
        </Typography>
      </Grid>

    </Grid>
  );
};

HeaderConnectBox.propTypes = {
  socketState: PropTypes.string.isRequired,
  loggedState: PropTypes.string.isRequired,
  colorState: PropTypes.string.isRequired,
};


export default HeaderConnectBox;
