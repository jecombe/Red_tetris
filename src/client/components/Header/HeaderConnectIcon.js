import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  connectIcon: (props) => ({
    color: props.connected ? 'lime' : 'red',
  }),
});

const HeaderConnectIcon = (props) => {
  const { connected } = props;
  const classes = useStyles({ connected });

  return (
    <FiberManualRecordIcon fontSize="small" className={classes.connectIcon} />
  );
};

HeaderConnectIcon.propTypes = {
  connected: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  connected: state.app.connected,
});

export default connect(mapStateToProps)(HeaderConnectIcon);
