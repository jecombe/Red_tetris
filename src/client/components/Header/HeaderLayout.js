import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import HomeIcon from '@material-ui/icons/Home';
import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord';
import { makeStyles } from '@material-ui/core/styles';

import HeaderLogo from './HeaderLogo';

const useStyles = makeStyles({
  homeIcon: {
    color: 'grey',
    '&:hover': {
      color: 'red',
    },
  },
  connectIcon: {
    color: (props) => (props.connexion ? 'lime' : 'red'),
  },
});

const HeaderLayout = (props) => {
  const { connexion, handleHomeButton } = props;
  const classes = useStyles(connexion);

  return (
    <Grid container alignItems="center">
      <Grid item xs={3} container justify="center">
        <IconButton aria-label="Home" component="span" onClick={handleHomeButton} className={classes.homeIcon}>
          <HomeIcon />
        </IconButton>
      </Grid>
      <Grid item xs={6} container justify="center">
        <HeaderLogo />
      </Grid>
      <Grid item xs={3} container justify="center">
        <FiberManualRecordIcon className={classes.connectIcon} />
      </Grid>
    </Grid>
  );
};

HeaderLayout.propTypes = {
  connexion: PropTypes.bool.isRequired,
  handleHomeButton: PropTypes.func.isRequired,
};

export default HeaderLayout;
