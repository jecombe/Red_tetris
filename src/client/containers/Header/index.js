import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Grid from '@material-ui/core/Grid';

import HeaderHome from './HeaderHome';
import HeaderLogo from '../../components/Header/HeaderLogo';
import HeaderConnect from './HeaderConnect';

const Header = () => (
  <AppBar color="secondary">
    <Grid container alignItems="center" spacing={4}>
      <Grid item xs={3} lg={4}>
        <HeaderHome />
      </Grid>
      <Grid item xs={6} lg={4}>
        <HeaderLogo />
      </Grid>
      <Grid item xs={3} lg={4}>
        <HeaderConnect />
      </Grid>
    </Grid>
  </AppBar>
);

export default Header;
