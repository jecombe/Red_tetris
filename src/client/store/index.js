import { createStore, applyMiddleware } from 'redux';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import rootReducer from '../reducers';
import socketIoMiddleware from '../middleware/socketIoMiddleware';

const logger = createLogger();

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(
      thunk,
      socketIoMiddleware,
      logger,
    ),
  ),
);

export default store;
