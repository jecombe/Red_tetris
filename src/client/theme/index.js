import { createMuiTheme } from '@material-ui/core/styles';
import red from '@material-ui/core/colors/red';
import grey from '@material-ui/core/colors/grey';

const theme = createMuiTheme({
  palette: {
    primary: red,
    secondary: grey,
  },
  status: {
    danger: 'orange',
  },
});

export default theme;
