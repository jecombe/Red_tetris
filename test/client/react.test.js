import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';

import store from '../../src/client/store';
import theme from '../../src/client/theme';

import App from '../../src/client/containers/app';
import FooterLayout from '../../src/client/components/Footer/FooterLayout';
import HeaderLayout from '../../src/client/components/Header/HeaderLayout';
import Header from '../../src/client/containers/header';


describe('Footer', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <HashRouter hashType="noslash">
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <App />
          </ThemeProvider>
        </Provider>
      </HashRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('snapshot renders', () => {
    const component = renderer.create(<FooterLayout />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('snapshot renders', () => {
    const appTest = {
      connected: true,
      rooms: [],
    };
    const handleButtonTest = () => {

    };
    const component = renderer.create(
      <HeaderLayout
        app={appTest}
        handleHomeButton={handleButtonTest}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  // test('snapshot renders', () => {
  //   const component = renderer.create(
  //     <Header />,
  //   );
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
