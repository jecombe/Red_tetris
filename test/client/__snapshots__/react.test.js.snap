// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot renders 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-3"
    >
      <span
        aria-disabled={false}
        aria-label="Home"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-homeIcon-9"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-6"
    >
      <img
        alt="Logo"
        src="header1.png"
        width="40%"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-connectIcon-10 makeStyles-connectIcon-11 MuiSvgIcon-fontSizeSmall"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M24 24H0V0h24v24z"
          fill="none"
        />
        <circle
          cx="12"
          cy="12"
          r="8"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
      color="primary"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Made by 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://github.com/jecombe"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        jecombe
      </a>
       & 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://github.com/dz0nda"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        dzonda
      </a>
       • 
      2019
    </p>
  </div>
</div>
`;

exports[`Footer snapshot renders 2`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
  >
    Made by 
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
      href="https://github.com/jecombe"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      jecombe
    </a>
     & 
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
      href="https://github.com/dz0nda"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      dzonda
    </a>
     • 
    2019
  </p>
</div>
`;

exports[`Footer snapshot renders 3`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-3"
  >
    <span
      aria-disabled={false}
      aria-label="Home"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-homeIcon-333"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-6"
  >
    <img
      alt="Logo"
      src="header1.png"
      width="40%"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-3"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root makeStyles-connectIcon-334 makeStyles-connectIcon-335 MuiSvgIcon-fontSizeSmall"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M24 24H0V0h24v24z"
        fill="none"
      />
      <circle
        cx="12"
        cy="12"
        r="8"
      />
    </svg>
  </div>
</div>
`;
